source/Plugins.cpp

```diff
@@ -164,3 +187,78 @@ void Plugins::TogglePlugin(const string &name)
 	auto *plugin = plugins.Get(name);
 	plugin->currentState = !plugin->currentState;
 }
+
+
+
+future<void> Plugins::Install(const InstallData &installData, bool guarded)
+{
+	oldNetworkActivity = true;
+	if(!guarded)
+	{
+		lock_guard<mutex> guard(activePluginsMutex);
+		if(activePlugins.count(installData.name))
+			return future<void>();
+		else
+			activePlugins.insert(installData.name);
+
+		// Create a new entry for the plugin.
+		Plugin *newPlugin = plugins.Get(installData.name);
+		newPlugin->name = installData.name;
+		newPlugin->aboutText = installData.aboutText;
+		newPlugin->path = Files::Plugins() + installData.name + "/";
+		newPlugin->version = installData.version;
+		newPlugin->enabled = false;
+		newPlugin->currentState = true;
+	}
+	return async(launch::async, [installData]() noexcept -> void
+		{
+
+			bool success = PluginHelper::Download(installData.url,
+				Files::Plugins() + installData.name + ".zip");
+			if(success)
+			{
+				success = PluginHelper::ExtractZIP(
+					Files::Plugins() + installData.name + ".zip",
```

Comment:
You're repeating the name of the zip file multiple times in this lambda

Reply:
made a variable